gromacs/fileformats/top.py

```diff
@@ -0,0 +1,1243 @@
+                    print('Uknown section in topology: %s' % curr_sec)
+        
+        # process cmap_lines
+        curr_cons = None
+        for line in cmap_lines:
+
+            # cmaptype opening line
+            if len(line.split()) == 8:
+                cons = blocks.CMapType('gromacs')
+                
+                atype1, atype2, atype3, atype4, atype8, func, sizeX, sizeY = line.replace("\\","").split()
+                func, sizeX, sizeY = int(func), int(sizeX), int(sizeY)
+                cons.atype1 = atype1
+                cons.atype2 = atype2
+                cons.atype3 = atype3
+                cons.atype4 = atype4
+                cons.atype8 = atype8
+                cons.gromacs = {'param':[], 'func': func}
+
+                curr_cons = cons
+
+            # cmap body
+            elif len(line.split()) == 10:
+                cmap_param = map(float, line.replace("\\","").split())
+                cons.gromacs['param'] += cmap_param
+
+            # cmaptype cloning line
+            elif len(line.split()) == 6:
+                cmap_param = map(float, line.replace("\\","").split())
+                cons.gromacs['param'] += cmap_param
+                self.cmaptypes.append(curr_cons)
+            else:
+                raise ValueError
+
+
+
+class SystemToGroTop(object):
+
+    formats = {
+        'atomtypes'      : '{:<7s} {:3s} {:3d} {:>7.6f}   {:4.3f}   {:3s}     {:14.12f}     {:10.9f}  \n',
+        'atoms'          : '{:6d} {:>10s} {:6d} {:6s} {:6s} {:6d} {:f} {:11.4f} \n',
+        'atoms_nomass'   : '{:6d} {:>10s} {:6d} {:6s} {:6s} {:6d} {:f}\n',
+        'nonbond_params' : '{:20s}  {:20s}  {:1d}  {:10.5f}  {:10.5f}\n',
+        'bondtypes'      : '{:5s}  {:5s}  {:1d}  {:6.4f}  {:6.1f}\n',
+        'bonds'          : '{:3d}  {:3d}   {:1d}\n',
+        'settles'        : '{:3d}  {:3d}  {:11.5f} {:11.5f}\n',
+        'virtual_sites3' : '{:3d}  {:3d}  {:3d}  {:3d}   {:1d}  {:12.10f}  {:12.10f}\n',
+        'exclusions'     : '{:3d}  {:3d}  {:3d}\n',
+        'pairtypes'      : '{:6s} {:6s}   {:d}    {:14.12f}     {:14.12f}    \n',
+        'pairs'          : '{:3d} {:3d}   {:1d}\n',
+        'angletypes_1'   : '{:>8s} {:>8s} {:>8s} {:1d}    {:8.4f}    {:10.5f}\n',
+        'angletypes_5'   : '{:>8s} {:>8s} {:>8s} {:1d}    {:8.4f}    {:10.5f}    {:9.5f}    {:11.5f}\n',
+        'constrainttypes': '{:6s} {:6s} {:1d}    {:8.6f}\n',
+        'angles'         : '{:3d} {:3d} {:3d}   {:1d}\n',
+        'dihedraltypes'  : '{:6s} {:6s} {:6s} {:6s}   {:1d}    {:6.2f}    {:f}    {:1d}\n',
+        'dihedrals'      : '{:3d} {:3d} {:3d} {:3d}   {:1d}\n',
+        'dihedrals_ext'  : '{:3d} {:3d} {:3d} {:3d}   {:1d}    {:6.2f}    {:f}    {:1d}\n',
+        'impropertypes'  : '{:6s} {:6s} {:6s} {:6s}   {:1d} {:6.2f} {:8.4f} \n',
+        'impropers'      : '{:3d} {:3d} {:3d} {:3d}   {:1d}\n',
+        'impropers_ext'  : '{:3d} {:3d} {:3d} {:3d}   {:1d} {:6.2f} {:8.4f} \n',
+    }
+
+
+    toptemplate = ""
```

Comment:
I mean the code *below* – appending to `toptemplate` just so that the indentation remains??
Why not
```python
import textwrap

   toptemplate="""
         [ defaults ]
         *DEFAULTS*
         [ atomtypes ]
         *ATOMTYPES*
         ; etc etc
         """
   toptemplate = textwrap.dedent(toptemplate)
```